doto-frontend/src/components/Streak.js

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import streakImage from "./images/streak.png";
+
+class Streak extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { points: 0 };
+    }
+
+    incrementStreak(change) {
+        this.setState({
+            points: this.state.points + change,
+        });
+    }
+
+    resetStreak() {
+        this.setState({
+            points: 0,
+        });
+    }
+
+    updateStreak() {
+        const currentDateTime = new Date();
+        let latestDate = null;
+
+        // find the last, previous uncompleted task
+        this.props.tasks.forEach(task => {
+            if (task.endDate < currentDateTime) {
+                if (!task.isComplete) {
+                    if (!latestDate && latestDate < task.startDate) {
+                        latestDate = task.endDate;
+                    }
+                }
+            }
+        });
+
+        // Steak = sum the values of every task completed since then
+        this.resetStreak();
+        this.props.tasks.forEach(task => {
+            if (!currentDateTime || task.endDate < currentDateTime) {
+                if (task.startDate >= latestDate) {
+                    this.incrementStreak(1);
+                }
+            }
+        });
+    }
+
+    componentDidMount() {
+        this.interval = setInterval(() => this.updateStreak(), 1000);
```

Comment:
I just realised, this means the updateStreak method will be called every second. Which is quite inefficient. Surely we only need to call it once on mount, then again when a task status is updated.

Reply:
We can do, it just won't get updated if the time where a task should have been copleted passes.  In this case, the streak really should get set to 0 (because they failed to complete a task), but if no action is taken, the streak will not get updated.  
Alternatively, it could get called less regularly (once a minute) as you can only set the duration to be in minutes.

Reply:
There is an issue where streak counter also flashes from one value to another when completing a task. This may also be the cause of it.

Reply:
> We can do, it just won't get updated if the time where a task should have been copleted passes. In this case, the streak really should get set to 0 (because they failed to complete a task), but if no action is taken, the streak will not get updated.
> Alternatively, it could get called less regularly (once a minute) as you can only set the duration to be in minutes.
Ah I see, good thinking, yea I would say this should get called less regularly.